apps/aecontract/src/aect_contracts.erl

```diff
@@ -245,11 +251,13 @@ new(Owner, Nonce, CTVersion, Code, Deposit) ->
                    owner_id     = aeser_id:create(account, Owner),
                    ct_version   = CTVersion,
                    code         = Code,
+                   code_ref     = no_ref,
                    store        = aect_contracts_store:new(),
                    log          = <<>>,
                    active       = true,
                    referrer_ids = [],
-                   deposit      = Deposit
+                   deposit      = Deposit,
+                   version      = serialization_version_for_vm(CTVersion)
```

Comment:
You need to transform the state trees when enabling IRIS - btw. take a look at how we manage multiple versions of account nodes in account state trees.

Reply:
Why do I need to do this? This way I managed to avoid doing that.

Reply:
You have still old DB records present in the DB. The only reason why the CI suite didn't fail is that our DB smoke tests only test ~1k blocks - essentially no contracts were present back then.
How to crash your code:
1. Take a snapshot of mainnet(https://downloads.aeternity.io/)
2. Start the node
3. Observe as block insertion crashes as contract_call tx crash your code ;P

Reply:
It will not cause I made sure old serialization and deserialization is preserved. Actually your solution will not work as we can't change old serialization, becouse it would change hashes. Please look how serialization and deserailization is handled.